src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs

```diff
@@ -727,6 +729,16 @@ private bool IsOperationComplete()
 
         private void SetOperationComplete()
         {
+            // When sharing the testhost between discovery and execution we must keep the
+            // testhost alive after completing the operation it was spawned for. As such we
+            // suppress the test request sender channel close taking place here. This channel
+            // will be closed when the test session owner decides to dispose of the test session
+            // object.
+            if (!this.CloseConnectionOnOperationComplete)
```

Comment:
Yes, `DiscoveryComplete`/`RunComplete` do usually close the communication channel and we want to avoid that for testhost reuse. Same goes for `DiscoveryAborted`/`RunAborted`. Because an abort request came from TE doesn't mean the testhost(s) are inconsistent, they may be reused if TE decides to re-issue the discovery/run request, so it'd be better to keep the testhost(s) warm.

Reply:
as long as there are no internal abort requests fired from within TP, i.e. aborts are mostly a result of the test discovery/runs being cancelled then it makes sense to keep the testhosts warm

Reply:
Sounds good 👍